File: math/cpp/prime.cpp
// This is supposed to calculate a list of prime numbers
// but it turns out there isn't a consistent formula for finding new ones
// so this script is broken

#include <iostream>
#include <vector>

bool check_prime(unsigned int n) {
	for (int i = 1;i < n + 1;i++) {
		if (n % i == 0) {
			if (i != 1 && i != n) {
				return false;
			}
		}
	}
	return true;
}

std::vector < unsigned int > new_prime (unsigned int n,unsigned int xrange) {
	std::vector < unsigned int > list = {};
	unsigned int x = n;
	for (int i = 0;i < xrange + 1;i++) {
		x = 6*x+1;
		list.push_back(x);
	}
	x = n;
	for (int i = 0;i < xrange + 1;i++) {
		x = 6*x-1;
		list.push_back(x);
	}
	return list;
}

double find_percentage(unsigned int xrange) {
	std::vector < bool > list = {};
	for (int i = 1;i < 41;i++) {
		unsigned int n = i*i+i+41;
		list.push_back(check_prime(n));
		for (unsigned int item: new_prime(n,xrange)) {
			list.push_back(check_prime(item));
		}
	}
	double truecount = std::count(list.begin(),list.end(),true);
	double size = list.size();
	return truecount / size * 100;
}
	
int main() {
	std::cout << "The formula gets a prime number " << find_percentage(2) << "% of the time." << std::endl;
}

File: math/python/prime.py
def check_prime(n:int):
	return len([ i for i in range(1,n + 1) if n % i == 0 ]) == 2
	
def new_prime(n:int,xrange:int):
	list = []
	if not check_prime(n):
		return list
	x = n
	for i in range(xrange):
		x = 6*x+1
		list.append(x)
	x = n
	for i in range(xrange):
		x = 6*x-1
		list.append(x)
	return list

def find_primes(irange:int,xrange:int):
	list = []
	if irange > 40:
		irange = 40
	for i in range(irange):
		n = i*i+i+41
		list.append(n)
		list.extend(new_prime(n,xrange))
	return [ i for i in list if check_prime(i) ]

def main():
	recursion_depth = 1
	list = find_primes(40,recursion_depth)
	print(f"The formula has found {len(list)} prime numbers with a recursion depth of {recursion_depth}.")

if __name__ == "__main__":
    main()
